src/vs/workbench/parts/terminal/electron-browser/terminalService.ts

```diff
@@ -169,6 +175,20 @@ export class TerminalService extends AbstractTerminalService implements ITermina
 	}
 
 	private _detectWindowsShells(): TPromise<IPickOpenEntry[]> {
+		const expectedLocations = this._getExpectedLocations();
```

Comment:
I don't think we should change this function but rather include a list of possible WSL shell paths (wsl.exe or bash.exe https://github.com/Tyriar/vscode-terminal-here/issues/15#issuecomment-367450527) in its own constant.

Reply:
> include a list of possible WSL shell paths (wsl.exe or bash.exe Tyriar/vscode-terminal-here#15 (comment)) in its own constant. 

Do you mean we should instantiate the array inside of this function and add wsl.exe to it?